tests/unit/lms/services/application_instance_test.py

```diff
@@ -84,17 +84,42 @@ def test_get_by_deployment_id_raises_on_missing(
             (None, None),
         ],
     )
```

Comment:
Hard to leave a suggestion here as it wasn't in this PR, but we can let `pytest` do the combinations for us:

```suggestion
    @pytest.mark.parametrize("developer_secret", (None, "SECRET"))
    @pytest.mark.parametrize("developer_key", (None, "KEY"))
```